app/views/assignments/index.html.erb

```diff
@@ -16,29 +27,22 @@
       <%= print_date(week) %>
     </td>
     
-    <% week_assignments = week.employees.order(:first_name) %>
-    <% week_assignments.each do |employee| %>
+    <% week_assignments = get_weekly_office_assignments(week, @office) %>
+    <% week_assignments.each do |assignment| %>
       <td>
-        <%= "#{employee.first_name}" %>
+        <%= "#{assignment.employee.first_name}" %>
       </td>
     <% end %>
     
-    <% if week_assignments.count == 1 %>
-      <td>
-        <b>EMPTY</b>
-      </td>
-    <% elsif week_assignments.count == 0 %>
-      <td>
-        <b>EMPTY</b>
-      </td>
-      <td>
-        <b>EMPTY</b>
-      </td>
+    <% if week_assignments.count != 2 %>
+      <% (2 - week_assignments.count).times do %>
+        <td><b>EMPTY</b></td>
```

Comment:
I wish we had presenters! We could override the `assignments` method and top up the assignments with `NullAssignment` objects that know how to render themselves!

🚀